dss/operations/checkout.py

```diff
@@ -0,0 +1,183 @@
+                    verify_delete(handle, bucket, key)
+            elif FILE_PREFIX in _key:
+                # should handle other keys, files/blobs
+                file_metadata = verify_get(handle, replica.bucket, _key)
+                verify_delete(handle, bucket, key=compose_blob_key(file_metadata))
+
+
+@checkout.action("verify",
+                 arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
+                            "--keys": dict(nargs="+", help="keys to check the status of", required=False)})
+def verify(argv: typing.List[str], args: argparse.Namespace):
+    """
+    Verify that keys are in the checkout bucket
+    """
+    replica = Replica[args.replica]
+    handle = Config.get_blobstore_handle(replica)
+    bucket = replica.checkout_bucket
+    checkout_status = dict()
+    for _key in args.keys:
+        if DSS_BUNDLE_KEY_REGEX.match(_key):  # handles bundles/fqid keys or fqid
+            uuid, version = parse_key(_key)
+            bundle_manifest = get_bundle_manifest(replica=replica, uuid=uuid, version=version)
+            checkout_bundle_contents = [x[0] for x in handle.list_v2(bucket=bucket, prefix=f'bundles/{uuid}.{version}')]
+            bundle_internal_status = list()
+
+            for _file in bundle_manifest['files']:
+                temp = collections.defaultdict(blob_checkout=False, bundle_checkout=False, should_be_cached=False)
+                bundle_key = f'bundles/{uuid}.{version}/{_file["name"]}'
+                blob_key = compose_blob_key(_file)
+
+                blob_status = verify_blob_existance(handle, bucket, blob_key)
+                if blob_status:
+                    temp['blob_checkout'] = True
+                if bundle_key in checkout_bundle_contents:
+                    temp['bundle_checkout'] = True
+                if cache_flow.should_cache_file(_file['content-type'], _file['size']):
+                    temp['should_be_cached'] = True
+
+                for x in ['name', 'uuid', 'version']:
+                    temp.update({x: _file[x]})
+                bundle_internal_status.append(temp)
+            checkout_status[_key] = bundle_internal_status
+        elif FILE_PREFIX in _key:
+            temp = collections.defaultdict(blob_checkout=False, should_be_cached=False)
+            file_metadata = verify_get(handle, replica.bucket, _key)
+            blob_key = compose_blob_key(file_metadata)
+            blob_status = verify_blob_existance(handle, bucket, blob_key)
+            if blob_status:
+                temp['blob_checkout'] = True
+            if cache_flow.should_cache_file(_file['content-type'], _file['size']):
+                temp['should_be_cached'] = True
+
+            for x in ['name', 'uuid', 'version']:
+                temp.update({x: _file[x]})
+            checkout_status[_key] = collections.defaultdict(uuid=temp)
+    print(json.dumps(checkout_status, indent=4, sort_keys=True))
```

Comment:
Does setting `indent=4` and having Python print the JSON in a more readable format make the JSON invalid? I can't seem to remember.

Related to [this comment](https://github.com/HumanCellAtlas/data-store/pull/2264#issuecomment-519290408) from Brian.

Reply:
seems to work correctly for me.... 
```./scripts/dss-ops.py checkout verify --keys bundles/a78072f5-e12d-4c59-abac-e2dc2e618e35.2019-08-08T172842.596019Z --replica aws | jq
{
  "bundles/a78072f5-e12d-4c59-abac-e2dc2e618e35.2019-08-08T172842.596019Z": [
    {
      "blob_checkout": true,
      "bundle_checkout": true,
      "name": "homer_squish.gif",
      "should_be_cached": false,
      "uuid": "9b45c517-0068-4d5a-a83b-59319613ba4e",
      "version": "2019-08-08T172842.596019Z"
    },
    {
      "blob_checkout": true,
      "bundle_checkout": true,
      "name": "Kirbyhappyjump.gif~c200.gif",
      "should_be_cached": false,
      "uuid": "4635d5d7-de61-4f17-b191-4641a55e5e15",
      "version": "2019-08-08T172842.596019Z"
    }
  ]
}
``` 
i can remove it so the operator has to use jq to get a pretty print.

Reply:
Works for me too, either way. :)